src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/SignatureHelper.cs

```diff
@@ -294,6 +294,7 @@ private static void WriteSimpleSignature(SignatureTypeEncoder signature, Type ty
         }
```

Comment:
Late to the party, sorry...

An enum value for `ValueType` was added, but didn't add it to the switch so effectively this means that we'll fall through to the code at line [292](https://github.com/dotnet/runtime/pull/104055/files#diff-1f73a2b03e75f53401d7258176b8348caeb3f1f64146d1448be5236f3ed55a82R292) if it was the new `ValueType` case. 

Would it be clearer to add the case/default and **also** make clear what the meaning of a `null` returned at [line 232](https://github.com/dotnet/runtime/pull/104055/files#diff-1f73a2b03e75f53401d7258176b8348caeb3f1f64146d1448be5236f3ed55a82R232) means?

Reply:
@buyaa-n We should address this ASAP. It also seems we have a test hole.

Reply:
> @buyaa-n We should address this ASAP. It also seems we have a test hole.

There is no test hole or bug. Nothing says that WriteSimpleSignature must special case all `CoreTypeId` values. It is why CoreTypeId is a separate enum from `SignatureTypeCode`.

Reply:
(I do not mind if somebody wants to make the comments better.)

Reply:
> An enum value for ValueType was added, but didn't add it to the switch so effectively this means that we'll fall through to the code at line [292](https://github.com/dotnet/runtime/pull/104055/files#diff-1f73a2b03e75f53401d7258176b8348caeb3f1f64146d1448be5236f3ed55a82R292) if it was the new ValueType case.

Yes, `ValueType` case not added because it is not a real a core type that has special signature for it, that is why it was not there previously.  `ValueType` signature will be added just as all other "normal" type, on row 293

> Would it be clearer to add the case/default and also make clear what the meaning of a null returned at [line 232](https://github.com/dotnet/runtime/pull/104055/files#diff-1f73a2b03e75f53401d7258176b8348caeb3f1f64146d1448be5236f3ed55a82R232) means?

Default case will be same as `null` case, row 292-293, no need to repeat that, I can add a comment to clarify later 

> @buyaa-n We should address this ASAP. It also seems we have a test hole.

The new test cover the ValueType reference with the local variable, other scenarios could be added like method argument or return type, field type etc but they will cross same path in the code

Reply:
Oh, I see. `SignatureTypeCode` is actually the `ELEMENT_TYPE_*` and `CoreTypeId` is a mapping for that array look-up. The arbitrary order of `CoreTypeId` now makes more sense. Apologies.

Reply:
Makes sense... just as a style question, would it be clearer to move [the end code](https://github.com/dotnet/runtime/pull/104055/files#diff-1f73a2b03e75f53401d7258176b8348caeb3f1f64146d1448be5236f3ed55a82R287-R294) to the `switch` body something like this?

```
                case CoreTypeId.TypedReference:
                    signature.TypedReference();
                    return;

               default:  // null or CoreTypeId.ValueType or some other value
                   {
                        EntityHandle typeHandle = module.GetTypeHandle(type);
                        signature.Type(typeHandle, type.IsValueType);
                        return;
                   }
            }
        }
```

Reply:
It would make sense only if the `return;`s are change to `break;`s.

The function as written is using "early exits" style - handle special cases with early exits and then do the non-special work in the main flow. If you move the non-special work to `default:` case of the switch statement, the early exits do not make sense anymore.

Reply:
I concur... I could do a PR for just that change if you would like... or of course leave to @buyaa-n or you folks

Reply:
I floated a branch here https://github.com/dotnet/runtime/compare/main...IDisposable:dotnet-runtime:refactor_emit_signature_helper can make a PR if desired

Reply:
> I could do a PR for just that change if you would like... or of course leave to @buyaa-n or you folks

I am not sure if it is an improvement, so not planning to change it at least not by myself, if others see it's better fine

Reply:
PR here https://github.com/dotnet/runtime/pull/104220